django/lego/dns.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""DNS utiilities."""
+
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from git import Git, Repo
+
+from .settings import DNS_REPOSITORY_URL, SSH_IDENTITY_FILE
+
+REPOSITORY_BASE_URL = "@".join(DNS_REPOSITORY_URL.split("@")[:-1])
+REPOSITORY_BRANCH = DNS_REPOSITORY_URL.split("@")[-1]
+RECORD_CONTENT = "; GENERATED BY httpreq - DO NOT EDIT!"
+SSH_EXECUTABLE = f"ssh -i {SSH_IDENTITY_FILE}"
+
+
+def write_dns_record(dns_record: str) -> None:
+    """Write a DNS record following the canonical-is-dns-configs specs if it doesn't exist.
+
+    Args:
+        dns_record: the DNS record to add.
+    """
+    with TemporaryDirectory() as tmp_dir, Git().custom_environment(GIT_SSH_COMMAND=SSH_EXECUTABLE):
+        repo = Repo.clone_from(REPOSITORY_BASE_URL, tmp_dir, branch=REPOSITORY_BRANCH)
+        dns_record_file = Path(f"{repo.working_tree_dir}/{dns_record}")
```

Comment:
I'm wondering if we shouldn't have a dedicated folder for these files to avoid polluting the root folder

Reply:
The branch in the repository will only contain the entries managed via this app